src/java/com/puppetlabs/ssl_utils/SSLUtils.java

```diff
@@ -465,30 +464,24 @@ public static void validateCertChain(List<X509Certificate> certs,
         throws CertificateException, CertPathValidatorException, IOException,
                InvalidAlgorithmParameterException, KeyStoreException,
                NoSuchAlgorithmException, NoSuchProviderException {
-        // there are bouncycastle versions of CertificateFactory and CertPath
-        // that I haven't looked into - not sure when using bouncycastle vs. java
-        // makes sense
-        final CertificateFactory certFactory = CertificateFactory.getInstance("X.509");
-        final CertPath certPath = certFactory.generateCertPath(certs);
+        final CertificateFactory certFactory;
         final CertPathValidator validator;
+        final CertStore crlStore;
+        final CollectionCertStoreParameters storeParams = new CollectionCertStoreParameters(crls);
         if (isFIPS()) {
-            validator = CertPathValidator.getInstance(PKIX_KEYMANAGER_ALGO, "BCFIPS");
+            certFactory = CertificateFactory.getInstance("X.509", BOUNCYCASTLE_FIPS_PROVIDER);
+            validator = CertPathValidator.getInstance(PKIX_KEYMANAGER_ALGO, BOUNCYCASTLE_FIPS_PROVIDER);
+            crlStore = CertStore.getInstance("Collection", storeParams, BOUNCYCASTLE_FIPS_PROVIDER);
         } else {
+            certFactory = CertificateFactory.getInstance("X.509");
             validator = CertPathValidator.getInstance(CertPathValidator.getDefaultType());
+            crlStore = CertStore.getInstance("Collection", storeParams);
         }
+        final CertPath certPath = certFactory.generateCertPath(certs);
         final KeyStore truststore = certsToTrustStore(certs);
-        PKIXRevocationChecker revocationChecker = (PKIXRevocationChecker)validator.getRevocationChecker();
-        revocationChecker.setOptions(EnumSet.of(
-                PKIXRevocationChecker.Option.PREFER_CRLS,
-                PKIXRevocationChecker.Option.NO_FALLBACK));
         PKIXBuilderParameters params =
                 new PKIXBuilderParameters(truststore, new X509CertSelector());
-        params.addCertPathChecker(revocationChecker);
-        // tests pass when this is true, false, or not included at all..
-        //params.setRevocationEnabled(true);
```

Comment:
when you set this to `false` does the validation succeed in the tests (and the tests then fail)?

Reply:
oh, I should try this again now that I got rid of the revocation checker.

Reply:
okay yes, now that I've removed the revocation checker, setting this to `false` makes all my test cases valid, so the tests fail.

Reply:
Were you thinking I should explicitly set this to `true` now or do you think it's fine to let the default handle it?

Reply:
I think it's fine to leave it using the default handler, I just wanted to know that the default behavior was the same between BCFIPS' and Java's built in verifiers.